src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3284,11 +3284,12 @@ private TypeSymbol StackMergeType(BoundExpression expr)
                 case BoundKind.Conversion:
                     var conversion = (BoundConversion)expr;
                     var conversionKind = conversion.ConversionKind;
-                    Debug.Assert(conversionKind != ConversionKind.DefaultOrNullLiteral);
```

Comment:
it seems weird to assert this condition and then test this same condition in the 'if' immediately below.

Reply:
It is, but I hesitate to change it.

---
In reply to: [313659767](https://github.com/dotnet/roslyn/pull/37596#discussion_r313659767) [](ancestors = 313659767)